procmonAnalyze.js

```diff
@@ -334,65 +356,102 @@ async function drawData(data, diskify) {
   let totalTime = maxTime - minTime;
   let trackWidth = canvas.width / tracks.length;
   let rendererScale = canvas.height / totalTime;
+
+  if (!isRefocused) {
+    gOuterState = {
+      minTime,
+      maxTime,
+      totalTime,
+      unconstrainedData: aData,
+      unconstrainedDiskifyData: aData,
+    };
+  }
+
   gState = {
-    minTime,
-    maxTime,
-    tracks,
-    totalTime,
-    trackWidth,
-    rendererScale,
-    targetRendererScale: rendererScale,
-    diskmapScale,
-    targetDiskmapScale: diskmapScale,
-    readsByPath,
-    diskify,
-    maxLcn,
-    diskmapTranslate: 0,
-    targetDiskmapTranslate: 0,
-    rendererTranslate: 0,
-    targetRendererTranslate: 0,
-    mouseX: 0,
-    mouseY: 0,
-    timelineIndicators: [],
-    lcnReads: [],
-    lastHoveredRect: null,
-    selectedEntry: null,
-    activePath: null,
+      minTime,
+      maxTime,
+      tracks,
+      totalTime,
+      trackWidth,
+      rendererScale,
+      targetRendererScale: rendererScale,
+      diskmapScale,
+      targetDiskmapScale: diskmapScale,
+      readsByPath,
+      diskify,
+      maxLcn,
+      diskmapTranslate: 0,
+      targetDiskmapTranslate: 0,
+      rendererTranslate: 0,
+      targetRendererTranslate: 0,
+      mouseX: 0,
+      mouseY: 0,
+      timelineIndicators: [],
+      lcnReads: [],
+      lastHoveredRect: null,
+      selectedEntry: null,
+      activePath: null,
+      topMarkerCoordinate: 0,
+      bottomMarkerCoordinate: markersCanvas.height,
+      isRefocused: isRefocused,
```

Comment:
Do we need `isRefocused` as separate state, when it can always be recomputed as `gOuterState.minTime != gOuterState.constrainedTimeBegin`? We could add a helper method `function isRefocused() {...}` to help make it cleaner to read when we need it.